src/az390.java

```diff
@@ -2410,6 +2421,41 @@ else if (tz390.op_trace_type[index]==613)                          // #533
 
 
 
+/**
+ * Comparator for mnemonics to mimic EBCDIC collating sequence
+ * The mnemonic portion occupies the first 8 positions of the ReportEntry
+ * and contains only uppercase characters and digits.
+ * Digits should sort higher than alphabetic characters.
+ */
+public class ReportEntryComparator implements Comparator<String>                   // #568
+ {@Override                                                                        // #568
+  public int compare(String str1, String str2)                                     // #568
+   {// special case: entries consisting of a single space sort first!              // #568
+    if (str1.charAt(0) == ' ') return 1;  // blank entry always smaller            // #568
+    if (str2.charAt(0) == ' ') return -1; // blank entry always smaller            // #568
+    for (int i=0; i<7; i++)                                                        // #568
+     {char c1 = str1.charAt(i);                                                    // #568
+      char c2 = str2.charAt(i);                                                    // #568
+      if (c1 == c2) {} // continue on to next char                                 // #568
+      else // we found different characters                                        // #568
+       {if (c1 == ' ') return -1; // Mnemonic 2 is longer                          // #568
+        if (c2 == ' ') return 1;  // Mnemonic 1 is longer                          // #568
```

Comment:
Can they be the same length?

Reply:
No, that would mean we have identical mnemonics. Should never occur. If it does we'll have bigger problems than this.